ml4gw/waveforms/phenom_p.py

```diff
@@ -0,0 +1,731 @@
+            chi22,
+            chip,
+            phic,
+            M,
+            xi,
+            dist_mpc,
+        )
+
+        hp, hc = self.PhenomPCoreTwistUp(
+            fs,
+            hPhenomDs,
+            eta,
+            chi1_l,
+            chi2_l,
+            chip,
+            M,
+            angcoeffs,
+            Y2,
+            alphaNNLOoffset - alpha0,
+            epsilonNNLOoffset,
+        )
+        t0 = (diffRDphase.unsqueeze(1)) / (2 * PI)
+        phase_corr = torch.cos(2 * PI * fs * (t0)) - 1j * torch.sin(
+            2 * PI * fs * (t0)
+        )
+        M_s = (m1 + m2) * MTSUN_SI
+        phase_corr_tc = torch.exp(
+            -1j * fs * M_s.unsqueeze(1) * tc.unsqueeze(1)
+        )
+        hp *= phase_corr * phase_corr_tc
+        hc *= phase_corr * phase_corr_tc
+
+        c2z = torch.cos(2 * zeta_polariz).unsqueeze(1)
+        s2z = torch.sin(2 * zeta_polariz).unsqueeze(1)
+        hplus = c2z * hp + s2z * hc
+        hcross = c2z * hc - s2z * hp
+        return hplus, hcross
+
+    def PhenomPCoreTwistUp(
+        self,
+        fHz: TensorType,
+        hPhenom: TensorType,
+        eta: TensorType,
+        chi1_l: TensorType,
+        chi2_l: TensorType,
+        chip: TensorType,
+        M: TensorType,
+        angcoeffs: Dict[str, TensorType],
+        Y2m: TensorType,
+        alphaoffset: TensorType,
+        epsilonoffset: TensorType,
+    ) -> Tuple[TensorType, TensorType]:
+        assert angcoeffs is not None
+        assert Y2m is not None
+        f = fHz * MTSUN_SI * M.unsqueeze(1)  # Frequency in geometric units
+        q = (1.0 + torch.sqrt(1.0 - 4.0 * eta) - 2.0 * eta) / (2.0 * eta)
+        m1 = 1.0 / (1.0 + q)  # Mass of the smaller BH for unit total mass M=1.
+        m2 = q / (1.0 + q)  # Mass of the larger BH for unit total mass M=1.
+        Sperp = chip * (
+            m2 * m2
+        )  # Dimensionfull spin component in the orbital plane.
+        # S_perp = S_2_perp chi_eff = m1 * chi1_l + m2 * chi2_l
+        # effective spin for M=1
+
+        SL = chi1_l * m1 * m1 + chi2_l * m2 * m2  # Dimensionfull aligned spin.
+
+        omega = PI * f
+        logomega = torch.log(omega)
+        omega_cbrt = (omega) ** (1 / 3)
+        omega_cbrt2 = omega_cbrt * omega_cbrt
+        alpha = (
+            (
+                angcoeffs["alphacoeff1"] / omega.T
+                + angcoeffs["alphacoeff2"] / omega_cbrt2.T
+                + angcoeffs["alphacoeff3"] / omega_cbrt.T
+                + angcoeffs["alphacoeff4"] * logomega.T
+                + angcoeffs["alphacoeff5"] * omega_cbrt.T
+            )
+            - alphaoffset
+        ).mT
+
+        epsilon = (
+            (
+                angcoeffs["epsiloncoeff1"] / omega.T
+                + angcoeffs["epsiloncoeff2"] / omega_cbrt2.T
+                + angcoeffs["epsiloncoeff3"] / omega_cbrt.T
+                + angcoeffs["epsiloncoeff4"] * logomega.T
+                + angcoeffs["epsiloncoeff5"] * omega_cbrt.T
```

Comment:
Torch will [not support](https://pytorch.org/docs/stable/tensors.html#torch.Tensor.T) the `.T` subsequently. Can you check if the `.mT`/`.t()` can be used as a replacement here and in other places?